rlberry/agents/torch/sac/sac.py

```diff
@@ -0,0 +1,307 @@
+import torch
+import torch.nn as nn
+import logging
+
+import gym.spaces as spaces
+from rlberry.agents import AgentWithSimplePolicy
+from rlberry.agents.utils.memories import Memory
+from rlberry.agents.torch.utils.training import optimizer_factory
+from rlberry.agents.torch.utils.models import default_policy_net_fn
+from rlberry.agents.torch.utils.models import default_value_net_fn
+from rlberry.agents.torch.utils.models import default_twinq_net_fn
+from rlberry.utils.torch import choose_device
+from rlberry.wrappers.uncertainty_estimator_wrapper import UncertaintyEstimatorWrapper
+
+logger = logging.getLogger(__name__)
+
+
+class SACAgent(AgentWithSimplePolicy):
+    """
+    Soft Actor Critic Agent (WIP). So far it is just a copy of A2C
+
+    A2C, or Advantage Actor Critic, is a synchronous version of the A3C policy
+    gradient method. As an alternative to the asynchronous implementation of
+    A3C, A2C is a synchronous, deterministic implementation that waits for each
+    actor to finish its segment of experience before updating, averaging over
+    all of the actors. This more effectively uses GPUs due to larger batch sizes.
+
+    Parameters
+    ----------
+    env : Model
+        Online model with continuous (Box) state space and discrete actions
+    batch_size : int
+        Number of episodes to wait before updating the policy.
+    horizon : int
+        Horizon.
+    gamma : double
+        Discount factor in [0, 1].
+    entr_coef : double
+        Entropy coefficient.
+    learning_rate : double
+        Learning rate.
+    optimizer_type: str
+        Type of optimizer. 'ADAM' by defaut.
+    k_epochs : int
```

Comment:
Isn't this what is called horizon in other agent ? I have some troubles understanding which is which between horizon, k_epochs and batch_size. Maybe you could explain more what is horizon ?

Reply:
ah yes this stuff is not finished and working properly. It was just the beginning. should we commit here just when things work properly?

Reply:
No you can commit before it is ready, no worries. This way we can help if there are problems specific to this PR, you can ask questions about it here.